### ISSUE TITLE
FortiOS - Use scp backup instead of parsing the show full-configuration

### ISSUE THREAD

**Issue by Najihel** [netfishers-onl/Netshot#130]
Hello,

The parsing of configuration could be very slow and truncated on huge config (+200k lines).

Maybe it's possible to use the scp transfer of the `sys_config` to reduce the time and the load of a snapshot more or less like the CheckPoint driver.

[Technical Note: How to download a FortiGate configuration file and upload firmware file using secure file copy (SCP)
](https://kb.fortinet.com/kb/documentLink.do?externalID=FD43754)

For example :
```
sshpass -p <PASSWORD> scp -q -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o ConnectTimeout=20 <USER>@<IP>:sys_config
```

Regards

**Comment by pizu** [comment]
Hi @SCadilhac 

We are noticing some discrepancies in the backup taken from web to cli.

The show command doesn't show the full configuration not even the show full-configuration. The link provided by @Najihel  is the only way to have the configuration, if you want I  can provide an output to compare to help you in this.

Thanks.

**Comment by pizu** [comment]
Hi,

Some examples:

in policy rules we are missing the below:
![image](https://user-images.githubusercontent.com/7047326/122830249-5f6a9c00-d2e8-11eb-851e-38dfe1e9a79b.png)

in some of the fqdn names we are finding "?" at the beginning
![image](https://user-images.githubusercontent.com/7047326/122830573-d011b880-d2e8-11eb-9953-09f3da7eef3e.png)

in some subnets names we are finding "?" at the end
![image](https://user-images.githubusercontent.com/7047326/122830680-f9324900-d2e8-11eb-83ea-bde3c9df53e3.png)

Thanks.

**Comment by SCadilhac** [comment]
Implemented in 9c0d6a40e157e6dc0e24320c613f5660515b16df
Requires next version of Netshot which will include cca878e84d7a1203e056b799bde7cd80f0465e07
